### PR TITLE
Raise a better error when a generic alias is used inside `type[]`

### PR DESCRIPTION
**Pull request by Viicos**
As per https://github.com/pydantic/pydantic/issues/11000#issuecomment-2535932284.

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: pydantic/_internal/_generate_schema.py [hunk 1]
```diff
@@ -1647,7 +1647,12 @@ def _subclass_schema(self, type_: Any) -> core_schema.CoreSchema:
         else:
             if _typing_extra.is_self(type_param):
                 type_param = self._resolve_self_type(type_param)
-
+            if _typing_extra.is_generic_alias(type_param):
+                raise PydanticUserError(
+                    'Subscripting `type[]` with an already parametrized type is not supported. '
+                    f'Instead of using type[{type_param!r}], use type[{_repr.display_as_type(get_origin(type_param))}].',
```
[the remark this answers is not in the document]
Comment (sydney-runkle) [reply pydantic/_internal/_generate_schema.py:1653]: This looks good. @dmontagu mentioned that we might want to support this eventually, so can we say "not yet supported", and link to the issue?

[the remark this answers is not in the document]
Comment (Viicos) [reply pydantic/_internal/_generate_schema.py:1653]: Imo trying to support this is going to introduce lots of maintenance burden for the reason I mentioned in https://github.com/pydantic/pydantic/issues/11000#issuecomment-2527730758, and maybe not possible at all as mentioned in https://github.com/pydantic/pydantic/issues/10462#issuecomment-2367301169.

[the remark this answers is not in the document]
Comment (sydney-runkle) [reply pydantic/_internal/_generate_schema.py:1653]: Fair enough, should we close #11000 then?

**Review by sydney-runkle — APPROVED** [review]
Thanks!

**Pull request merged**